src/wallet/wallet.h

```diff
@@ -683,6 +683,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
      * Generate a new key
      */
     CPubKey GenerateNewKey();
+    CKey DeriveNewKey(CKeyMetadata &metadata);
```

Comment:
nit: name is awfully close to GenerateNewKey to me. Perhaps something like "DeriveNewHDKey" or "DeriveNewChildKey".